IdentityCore/tests/MSIDKeyOperationUtilTest.m

```diff
@@ -0,0 +1,251 @@
+    
+    // Public key tests
+    
+    // Testing public key can be used to encrypt (kSecAttrCanEncrypt of public key set to YES)
+    XCTAssertTrue([[MSIDKeyOperationUtil sharedInstance] isOperationSupportedByKey:kSecKeyOperationTypeEncrypt algorithm:kSecKeyAlgorithmECIESEncryptionCofactorX963SHA256AESGCM key:self.eccPublicKey context:nil error:&error]);
+    // Testing public key cannot be used to decrypt (kSecAttrCanDecrypt of public key set to NO)
+    XCTAssertFalse([[MSIDKeyOperationUtil sharedInstance] isOperationSupportedByKey:kSecKeyOperationTypeDecrypt algorithm:kSecKeyAlgorithmECIESEncryptionCofactorX963SHA256AESGCM key:self.eccPublicKey context:nil error:&error]);
+    // Any public key should not be able to sign an artifact
+    XCTAssertFalse([[MSIDKeyOperationUtil sharedInstance] isOperationSupportedByKey:kSecKeyOperationTypeSign algorithm:kSecKeyAlgorithmECDSASignatureDigestX962SHA256 key:self.eccPublicKey context:nil error:&error]);
+    XCTAssertFalse([[MSIDKeyOperationUtil sharedInstance] isOperationSupportedByKey:kSecKeyOperationTypeSign algorithm:kSecKeyAlgorithmECDSASignatureDigestX962SHA256 key:self.rsaPublicKey context:nil error:&error]);
+    
+    // Operation not supported for NULL key
+    SecKeyRef nullKey = NULL;
+    XCTAssertFalse([[MSIDKeyOperationUtil sharedInstance] isOperationSupportedByKey:kSecKeyOperationTypeSign algorithm:kSecKeyAlgorithmRSASignatureRaw key:nullKey context:nil error:&error]);
+}
+
+- (void)testCorrectJwtalgorithmIsReturnedForAKey
+{
+    XCTAssertNotEqual(self.eccPrivateKey, NULL);
+    // Return ES256 for ECC private key
+    XCTAssertEqual([[MSIDKeyOperationUtil sharedInstance] getJwtAlgorithmForKey:self.eccPrivateKey context:nil error:nil], MSID_JWT_ALG_ES256);
+    NSError *error;
+    // Passing public key should not return a Jwt alg
+    XCTAssertNil([[MSIDKeyOperationUtil sharedInstance] getJwtAlgorithmForKey:self.eccPublicKey context:nil error:&error]);
+    XCTAssertNotNil(error);
+}
+
+- (void)testSigningRawDataWithKey
+{
+    NSData *dataToBeSigned = [@"TEST" dataUsingEncoding:NSUTF8StringEncoding];
+    NSData *dataDigest = [dataToBeSigned msidSHA256];
+    NSError *error;
+    NSData *signature = nil;
+    
+    // Valid signature with ECC private key via secure enclave
+    signature = [[MSIDKeyOperationUtil sharedInstance] getSignatureForDataWithKey:dataToBeSigned privateKey:self.eccPrivateKey context:nil error:&error];
+    XCTAssertNotNil(signature);
+    XCTAssertNil(error);
+
+    CFErrorRef verifyingError = NULL;
+    BOOL isVerified = SecKeyVerifySignature(self.eccPublicKey,
+                                            kSecKeyAlgorithmECDSASignatureMessageX962SHA256,
+                                            (__bridge CFDataRef) dataToBeSigned,
+                                            (__bridge CFDataRef) signature,
+                                            &verifyingError);
+    XCTAssertTrue(isVerified);
+    XCTAssertTrue(verifyingError == NULL);
```

Comment:
better to use XCTAssertNotNil

Reply:
Yeah, I had it that way before but XCTAssertNotNil(NULL) returns false instead of true.